scene/main/scene_tree.cpp

```diff
@@ -610,23 +610,29 @@ bool SceneTree::process(double p_time) {
 	if (Engine::get_singleton()->is_editor_hint()) {
 		//simple hack to reload fallback environment if it changed from editor
 		String env_path = GLOBAL_GET(SNAME("rendering/environment/defaults/default_environment"));
-		env_path = env_path.strip_edges(); //user may have added a space or two
-		String cpath;
-		Ref<Environment> fallback = get_root()->get_world_3d()->get_fallback_environment();
-		if (fallback.is_valid()) {
-			cpath = fallback->get_path();
-		}
-		if (cpath != env_path) {
-			if (!env_path.is_empty()) {
-				fallback = ResourceLoader::load(env_path);
-				if (fallback.is_null()) {
-					//could not load fallback, set as empty
-					ProjectSettings::get_singleton()->set("rendering/environment/defaults/default_environment", "");
+		// Ensure the uid path is mapped to a resource, which could not be
+		// the case if the editor is still scanning the filesystem.
+		ResourceUID::ID id = ResourceUID::get_singleton()->text_to_id(env_path);
```

Comment:
This won't work if the value is stored as path (which can happen if it was set before dev5). Though seems like environment will display correctly in either case, so not sure what does it do.

Reply:
It's because on scene tree initialization it calls `ResourceLoader::load` and it works.. I supposed it uses the `uid_cache.bin` in `.godot` or something. I will just reuse it.

https://github.com/godotengine/godot/blob/bbc54692c05ff6a85a6aeefbf5b9c87de91983d8/scene/main/scene_tree.cpp#L1941-L1972